src/app/students/(hooks)/useStudents.tsx

```diff
@@ -10,29 +11,54 @@ type DropdownOption  = {
 type StudentsContextType = {
   displayStudents: Student[];
   currentSections: DropdownOption[];
+  totalStudents: number;
 }
-
 const StudentsContext = createContext<StudentsContextType | undefined>(undefined)
 
 const useStudentsProvider = (): StudentsContextType => {
-    const [studentsParse, setStudentsParse] = useState<Student[]>([])
     const [displayStudents, setDisplayStudents] = useState<Student[]>([])
     const [sections, setSections] = useState<DropdownOption[]>([])
+    const [totalStudents, setTotalStudents] = useState<number>(0)
+    const searchParams = useSearchParams()
+
+    const { pageIndex, pageSize } = useMemo(() => {
+        return {
+            pageIndex: parseInt(searchParams.get('page') ?? '1', 10),
+            pageSize: parseInt(searchParams.get('per_page') ?? '10', 10),
+        }
+    }, [searchParams])
+
 
+    useEffect(() => {
+        const fetchStudents = async () => {
+            const fixedCourseNum = 1 // Replace with actual course number
+            const courseMemberData = await fetch(`http://127.0.0.1:8000/api/v1/course-members/?course=${fixedCourseNum}&page=${pageIndex}&per_page=${pageSize}`).then(res => res.json())
```

Comment:
the base url of the API should be an environment variable. Naturally it will not be localhost when deployed